gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -415,10 +415,33 @@ private Entry getEntry(int affinity) {
     return localEntries.get(index);
   }
 
+  private static class RetainedEntry {
+    private final Entry entry;
+    private final AtomicBoolean wasReleased;
+
+    public RetainedEntry(Entry entry) {
+      this.entry = entry;
+      wasReleased = new AtomicBoolean(false);
+    }
+
+    void release() {
+      if (!wasReleased.compareAndSet(false, true)) {
+        Exception e = new IllegalStateException("Entry was already released");
```

Comment:
This is another stacktrace we don't want to log, can we just log the WARNING without a stacktrace?
Or if we think this is a true error, then we can throw the Exception?

Reply:
We definitely need a stacktrace here, it will tell us how the refcount got negative. Otherwise we are no wiser than before.....we are just hiding the problem under the rug. And in this case alert firing woud be correct

Reply:
To me `Entry was already released` is a true error, why are we just logging instead of throwing an exception?

Reply:
Because throwing will bubble up to the enduser, who had nothing to do with the error. Instead I want to keep it localized to the double release and protect the rest of the application